packages/frontend/src/components/LimitButton/LimitForm.tsx

```diff
@@ -0,0 +1,75 @@
+import { Intent } from '@blueprintjs/core';
+import { Classes } from '@blueprintjs/popover2';
+import React, { FC, useCallback, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { useApp } from '../../providers/AppProvider';
+import { isOnlyNumbers } from '../../utils/fieldValidators';
+import Input from '../ReactHookForm/Input';
+import { Props } from './index';
+import { ApplyButton, StyledForm } from './LimitButton.styles';
+
+const LimitForm: FC<Pick<Props, 'limit' | 'onLimitChange'>> = ({
+    limit,
+    onLimitChange,
+}) => {
+    const { health } = useApp();
+    const methods = useForm<{ limit: string }>({
+        mode: 'onChange',
+    });
+    const {
+        setValue,
+        formState: { isValid },
+    } = methods;
+
+    useEffect(() => {
+        setValue('limit', limit?.toString());
+    }, [limit, setValue]);
+
+    const handleSubmit = useCallback(
+        (data: { limit: string }) => {
+            console.log('handleSubmit');
+            onLimitChange(parseInt(data.limit));
+        },
+        [onLimitChange],
+    );
+
+    if (!health.data) {
+        return null;
+    }
+    return (
+        <StyledForm
+            name="query_limit"
+            methods={methods}
+            onSubmit={handleSubmit}
+        >
+            <Input
```

Comment:
why not using a numeric input ?

Reply:
There is a really annoying behaviour when editing a number in the numeric input.
Eg: limit is 500 but you want 600. So the user deletes the 5 to then type 6. But the input instantly converts the `00` to 0 to be a valid number.